addons/mail/tests/discuss/test_discuss_channel.py

```diff
@@ -593,6 +593,25 @@ def test_channel_write_should_send_notification(self):
         ):
             channel.name = "test test"
 
+    def test_channel_mention_should_send_message_in_channel(self):
+        channel = self.env['discuss.channel'].create({"name": "Test", "channel_type": "channel"})
+        partner = self.env["res.partner"].create({
+            "name": "Test Partner",
+            "email": "testpartner@odoo.com",
+        })
+        # Message Post
+        message = partner.message_post(body="#Test", channel_ids=[channel.id], message_type="comment", subtype_xmlid="mail.mt_comment")
+        # Check that the mentioned channel containt a message with id of channel
+        message_in_db = self.env['mail.message'].search([('res_id', '=', channel.id), ('model', '=', 'discuss.channel')])
+        self.assertEqual(len(message_in_db), 1)
+        self.assertEqual(self.env['mail.message'].search([('id', '=', message.id), ('model', '=', 'res.partner')]).body, '<p>#Test</p>')
+        user = f'<span><a href="#" data-oe-model="res.partner" data-oe-id="{self.env.user.partner_id.id}">{self.env.user.partner_id.name}</a>'
+        backlink_html = f'<a href="#" data-oe-type="notification" data-oe-id="{partner.id}" data-oe-model="res.partner">Test Partner</a></span>'
+        self.assertEqual(
+            message_in_db.body,
+            f"{user} mentioned this channel in {backlink_html}"
+        )
```

Comment:
That's way too verbose for such a simple test.
1. `email` looks useless, so drop
2. Comments like `# Message Post` are useless too, the statement is obvious with function named `message_post`
3. Asserting that the message posted produce `#Test` textually has no value: the body is forged with `#Test` and this is definitely overly tested already
4. Not a python expert, but could we just assert text content here? The exact link with its attributes is not really useful, this adds more complexity than necessary. And if we want to test that click on link goes to the message, we don't care on the attributes: we care by testing actually click on this part of message and see that it opens the thread at the right message